CIX:Add OEM-specific ACPI OEM_ID check for default domain

Ghostor8 opened the pull request:
improves system bring-up reliability and correct domain assignment on CIXTEK-based platforms

## Summary by Sourcery

Adjust default IOMMU domain selection based on ACPI OEM_ID for specific platforms.

Bug Fixes:
- Ensure correct default IOMMU domain assignment on CIXTEK ACPI-based platforms by honoring OEM-specific identification.

Enhancements:
- Introduce a helper to check ACPI OEM_ID from the IORT table and use it to select a DMA default domain on matching systems.

Cryolitia reviewed (approved):
LGTM

Cryolitia commented on drivers/iommu/arm/arm-smmu-v3/arm-smmu-v3.c:
```diff
@@ -2880,6 +2880,32 @@ static int arm_smmu_dev_disable_feature(struct device *dev,
+			return 1;
+
+	}
+	return 0;
```
This suggestion should be applied

Cryolitia commented on drivers/iommu/arm/arm-smmu-v3/arm-smmu-v3.c:
```diff
@@ -2880,6 +2880,32 @@ static int arm_smmu_dev_disable_feature(struct device *dev,
+			return 0;
+		}
+
+		if (!strncmp(iort_table->oem_id, oem_id, 6))
```
Avoid using magic number, at lease a line of comment

MingcongBai reviewed (changes requested):
补丁机理没毛病，但我个人认为这个补丁应该拆成两个，否则会存在显著的描述混杂。观察补丁内容，我们会发现补丁做了两件事情：

1. 新增 ACPI OEM_ID 的检查功能（这是个特性）
2. 基于 OEM_ID 检查，针对此芯平台设置 IOMMU_DOMAIN_DMA 属性（这是个平台规避）

将平台规避描述成新特性可能不是作者的意图，但最终得到的是一个记录不清的补丁。

而且，我建议仔细描述到底出了什么问题，解决了什么问题，否则后续出现回归或类似问题将难以厘清/归纳原因和现象。

Commit:
ACPI: Add helper to check OEM_ID in ACPI IORT table

adds a utility function acpi_check_oem_id() to detect whether
the system firmware identifies a specific OEM_ID in the ACPI IORT table

Signed-off-by: wenxue.ding <Wenxue.Ding@cixtech.com>
Signed-off-by: Dylan.Wu" <Dylan.Wu@cixtech.com>

Commit:
iommu: Add CIXTEK-specific DMA domain handling based on ACPI OEM_ID

extends the PCI device domain selection logic by introducing a
vendor-specific path for CIXTEK platforms.

Signed-off-by: wenxue.ding <Wenxue.Ding@cixtech.com>
Signed-off-by: Dylan.Wu" <Dylan.Wu@cixtech.com>

Commit:
Update defconfig and cdns3 USB driver settings

updates align the USB controller settings with current system
requirements and enhance compatibility and stability for USB gadget
operations.

Signed-off-by: Dylan.Wu" <Dylan.Wu@cixtech.com>

opsiff commented on drivers/soc/cix/acpi/acpi_resource_lookup.c:
```diff
@@ -174,8 +174,11 @@ static int reset_lookup_handle(const union acpi_object *obj, void *data)
 
 static int rmem_dev_set_dma(struct device *dev, phys_addr_t base, size_t size)
 {
+dev_err(dev, "==== dev[%s], dev->dma[%s]", dev ? "1": "NULL", dev->dma_mem ? "1": "NULL");
```
这个日志信息级别不应该是dev_err 缩进有问题
